lib/internal/make-api-call.js

```diff
@@ -156,10 +175,15 @@ exports.inject = function(options) {
       throw 'Missing either a valid API key, or a client ID and secret';
     }
 
+    var isPost = method == 'POST';
     var requestUrl = url.format({
       pathname: path,
-      query: query
+      query: isPost ? {key: query.key} : query
     });
+    if (isPost) {
+      // Remove key from query for HTTP POST, since query is the JSON being posted.
+      delete query.key;
```

Comment:
I think we need POST logic in here, since the function is building the URL and the logic for that changes based on POST vs GET - but I missed the fact the API key gets added to the query object right in this functions, so I can definitely make this clearer by putting the logic there - namely we append the API key to the query object for GET requests, otherwise we just set the local query variable to {key: key} for POST requests.